test/parallel/test-stream-readableListening-state.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+/**
+ * Testing Readable Stream internal state
+ * `readableListening`
+ */
+
+const common = require('../common.js');
+const assert = require('assert');
+const stream = require('stream');
+
+/**
+ * Fiest test escenary, were we are going to call  the
+ * `readable` event and test that `readableListening` state
+ *  change succefully change to `true`
+ */
+
+const r = new stream.Readable({
+  read: () => {}
+});
+
+// readableListening state should start in `false`.
+assert.deepEqual(r._readableState.readableListening, false);
+
+r.on('readable', common.mustCall(() => {
+  // Inside the readable event this state should be true.
+  assert.deepEqual(r._readableState.readableListening, true);
+}));
+
+// This should trigger some events in our
+// first test ecenary
+r.push(Buffer.from('Testing readableListening state'));
+
+/**
+ * Second test escenary, where we never
+ * call the `readable` event and we expect
+ * that readableListening mantein his `false`
+ * value
+ */
+
+const r2 = new stream.Readable({
```

Comment:
Instead of having `r`, `r2`, etc., you can place the isolated test cases inside a block scope and reuse the same variable names.